packages/destination-actions/src/destinations/stackadapt-audiences/forwardProfile/__tests__/index.test.ts

```diff
@@ -0,0 +1,116 @@
+import nock from 'nock'
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
+import Definition from '../../index'
+import { SegmentEvent } from '@segment/actions-core/*'
+
+const testDestination = createTestIntegration(Definition)
+const mockGqlKey = 'test-graphql-key'
+
+const gqlHostUrl = 'https://sandbox.stackadapt.com'
+const gqlPath = '/public/graphql'
+const mockEmail = 'admin@stackadapt.com'
+const mockUserId = 'user-id'
+const mockEmail2 = 'email2@stackadapt.com'
+const mockUserId2 = 'user-id2'
+
+const defaultEventPayload: Partial<SegmentEvent> = {
+  userId: mockUserId,
+  type: 'identify',
+  traits: {
+    email: mockEmail,
+    first_time_buyer: true
+  },
+  context: {
+    personas: {
+      computation_class: 'audience',
+      computation_key: 'first_time_buyer'
+    }
+  }
+}
+
+const batchEventPayload: Partial<SegmentEvent> = {
+  userId: mockUserId2,
+  type: 'identify',
+  traits: {
+    email: mockEmail2
+  }
+}
+
+describe('forwardProfile', () => {
+  it('should translate audience entry/exit into GQL format', async () => {
+    let requestBody
+    nock(gqlHostUrl)
+      .post(gqlPath, (body) => {
+        requestBody = body
+        return body
+      })
+      .reply(200, { data: { success: true } })
+    const event = createTestEvent(defaultEventPayload)
+    const responses = await testDestination.testAction('forwardProfile', {
+      event,
+      useDefaultMappings: true,
+      settings: { apiKey: mockGqlKey }
+    })
+    expect(responses.length).toBe(1)
+    expect(responses[0].status).toBe(200)
+    expect(responses[0].request.headers).toMatchInlineSnapshot(`
+      Headers {
+        Symbol(map): Object {
+          "authorization": Array [
+            "Bearer test-graphql-key",
+          ],
+          "content-type": Array [
+            "application/json",
+          ],
+          "user-agent": Array [
+            "Segment (Actions)",
+          ],
+        },
+      }
+    `)
+    expect(requestBody).toMatchInlineSnapshot(`
+      Object {
+        "query": "mutation {
+            upsertProfiles(
+              subAdvertiserId: 1,
+              externalProvider: \\"Segment\\",
+              profiles: [{userId:\\"user-id\\",listId:\\"first_time_buyer\\",action:\\"enter\\",email:\\"admin@stackadapt.com\\"}]
```

Comment:
we wouldn't be taking in external audience info as `list` from Segment. We'll be parsing audience metadata from profile based on fields with prefix `audience_` so we can give audience id field name `audience_id` and name as `audience_name`.
Also to confirm that the profile attributes will be in camel case, right? I'll reflect that in the upsertProfile implementation

Reply:
I've updated the code to use `audience_id` and `audience_name` and to use snake case as your example.

For casing, we can use any case you want but we will need to write code to enforce the casing since we can't guarantee the casing the customer will use for trait keys. I'll do this in a subsequent PR if you let me know the casing you prefer.